provider/marathon_test.go

```diff
@@ -1434,3 +1447,76 @@ func TestMarathonGetSubDomain(t *testing.T) {
 		}
 	}
 }
+
+func TestGetBackendServer(t *testing.T) {
+
+	applications := []struct {
+		forceTaskHostname bool
+		application       marathon.Application
+		expected          string
+	}{
+		{
+			application: marathon.Application{
+				ID: "app-without-IP-per-task",
+			},
+			expected: "sample.com",
+		},
+		{
+			application: marathon.Application{
+
+				ID: "app-with-IP-per-task",
+				IPAddressPerTask: &marathon.IPAddressPerTask{
+					Discovery: &marathon.Discovery{
+						Ports: &[]marathon.Port{
+							{
+								Number: 8880,
+								Name:   "p1",
+							},
+						},
+					},
+				},
+			},
+			expected: "192.168.0.1",
+		}, {
+			forceTaskHostname: true,
+			application: marathon.Application{
+				ID: "app-with-hostname-enforced",
+				IPAddressPerTask: &marathon.IPAddressPerTask{
+					Discovery: &marathon.Discovery{
+						Ports: &[]marathon.Port{
+							{
+								Number: 8880,
+								Name:   "p1",
+							},
+						},
+					},
+				},
+			},
+			expected: "sample.com",
+		},
+	}
+
+	for _, app := range applications {
+		t.Run(fmt.Sprintf("running %s", app.application.ID), func(t *testing.T) {
```

Comment:
Can we parallelize the tests? If so, please call `t.Parallel()` here and do `app := app` before the call to `t.Run` to capture the loop variable.